Bootstrapped app with typescript

nikhil-thakkar opened the pull request:
Please review.

nikhil-thakkar committed:
Bootstrapped app with typescript

karntrehan commented:
Can you provide a bit of a background to this?

nikhil-thakkar commented:
This is a sample app created using react native cli with typescript as a language. This can be the starting point for others to start contributing.

nikhil-thakkar committed:
Updated the readme for installation instructions

nikhil-thakkar committed:
Updated build tools version for androidx

AniketSK commented on react-native/App/package.json:
```diff
@@ -21,8 +21,5 @@
     "metro-react-native-babel-preset": "0.48.5",
     "react-test-renderer": "16.6.3",
     "typescript": "3.2.2"
-  },
```
Why is jest removed? @nikhil-thakkar

AniketSK committed:
Updated react scripts to address critical vulnerability in previous versions.

AniketSK committed:
Added support for ts-jest testing and a sample test.

Only the below tutorials were followed:
1.
https://facebook.github.io/react-native/blog/2018/05/07/using-typescript-with-react-native
2. https://kulshekhar.github.io/ts-jest/user/react-native/

AniketSK committed:
Allowing for some versions to be updated automatically.
